docs/relationship_synthesis.md

```diff
@@ -0,0 +1,295 @@
+
+In order to create a new relationship synthesis rule, we need to follow the given steps:
+
+1. Create a `<ENTITY_TYPE-to-ENTITY_TYPE>.yml` file under the `relationships/synthesis` directory.
+   1. In this directory, you'll find all the existing relationship synthesis rules, 
+   and it's also an excellent source of inspiration for crafting new ones.
+
+Example: 
+
+```
+entity-definitions/
+| - definitions/ (entityTypes)
+  ...
+| - relationships/
+  | - candidates/
+  | - synthesis/
+    | EXT-SERVICE-to-EXT-PIXIE-DNS.yml
+    ...
+```
+
+2. Fill in the required fields (`name`, `version`, `origins`, `conditions`, `relationship`)
+following the instructions on the [next section](#how-to-configure-a-new-candidate-relationship).
+
+3. Create a new pull request and make sure all the automatic validations are successfully executed.
+
+4. Wait for our team to review the pull request and iterate on the feedback.
+
+5. Once approved, merged to the main branch and a new release is issued, it is available for you to use.
+
+6. Enjoy your new relationship synthesis rule being applied automatically to your data! :tada:
+
+# How to configure a new Candidate Relationship
+
+Relationship synthesis involves a two-phase system to determine and create relationships based on the provided rules.
+
+In the first phase, the system verifies if a data point should be considered for creating a relationship by performing duck typing.
+This means checking if the data point has all the required attributes and values specified by the rules.
+
+In the second phase, the data point is used to create a relationship according to the defined rule,
+which specifies the relationship type (guid to guid, candidate, or proposal)
+and how the source and target GUIDs should be created.
+
+Example of a valid relationship definition:
+
+```yaml
+relationships:
+  - name: extServiceCallsExtPixieDns
+    version: 1
+    origins:
+      - OpenTelemetry
+    conditions:
+      - attribute: eventType
+        anyOf: [ "MetricRaw" ]
+      - attribute: entity.type
+        anyOf: [ "PIXIE_DNS" ]
+    relationship:
+      expires: P75M
+      relationshipType: CALLS
+      source:
+        buildGuid:
+          account:
+            attribute: accountId
+          domain:
+            value: EXT
+          type:
+            value: SERVICE
+          identifier:
+            fragments:
+              - attribute: service.name
+            hashAlgorithm: FARM_HASH
+      target:
+        extractGuid:
+          attribute: entity.guid
+```
+
+In the provided example, you can observe the various sections of a valid relationship definition: `name`, `origins`, `conditions` and `relationship`. 
+Each of these sections is necessary for the rule to be considered valid.
+
+Below, we offer a more detailed explanation of how to complete each section properly.
+
+## Name
+
+The `name` property provides information during debugging and can be used for feature flags or logging purposes.
```

Comment:
Do you folks have any better ideas for a more user facing description? I imported it from our CDD, but all those details seem to be only internally applicable cc @naxhh @manuelrodriguez-nr

Reply:
Do we want a camelCase validation here too and uniqueness among all other names?

Discussion with product on the internal document: https://docs.google.com/document/d/1aF66Vrmoc-CkD0DuN5QkvTRzK4jqLjyijdA2Co2qHsA/edit?disco=AAAA1bx0dkw

Reply:
Changing the definition to be a unique identifier that follows the camelCase structure.